File: 6-1.py
#!/usr/bin/env python

answers = 0
for group in open('6.sample').read().split("\n\n"):
     group_answers = {}
     for person in group.split("\n"):
          for c in person.strip():
               group_answers[c] = True

     answers += len(group_answers)

print(f"{answers}")

File: 6-1.rb
#!/usr/bin/env ruby

answers = 0
File.read('6.input').split("\n\n").each do |group|
  group_answers = {}
  group.lines.each do |person|
    person.strip.each_char do |c|
      group_answers[c] = true
    end
  end
  answers += group_answers.count
end

print("#{answers}\n")
